html/modules/custom/reliefweb_import/src/Plugin/ReliefWebImporter/FaoDataImporter.php

```diff
@@ -0,0 +1,466 @@
+    $plugin->setPluginSetting('validate_file_content_type', FALSE);
+
+    // Max import attempts.
+    $max_import_attempts = $this->getPluginSetting('max_import_attempts', 3, FALSE);
+
+    // Retrieve the list of documents manually posted so we can exclude them
+    // from the import.
+    $ids = array_filter(array_map(fn($item) => $item['id'] ?? NULL, $documents));
+    $manually_posted = $this->getManuallyPostedDocuments(
+      $ids,
+      'https://data-in-emergencies.fao.org/documents/{id}',
+      '#^https://data-in-emergencies.fao.org/documents//[^/]+/(\d+)[^/]*$#i',
+    );
+
+    // Retrieve the list of existing import records for the documents.
+    $uuids = array_filter(array_map(fn($item) => $this->generateUuid($item['documentLink'] ?? ''), $documents));
+    $existing_import_records = $this->getExistingImportRecords($uuids);
+
+    // Prepare the documents and submit them.
+    $processed = 0;
+    $import_records = [];
+    foreach ($documents as $document) {
+      $import_record = [
+        'importer' => $this->getPluginId(),
+        'provider_uuid' => $provider_uuid,
+        'entity_type_id' => $entity_type_id,
+        'entity_bundle' => $bundle,
+        'status' => 'pending',
+        'message' => '',
+        'attempts' => 0,
+        'source' => 'FAO Data',
+      ];
+
+      // Retrieve the document ID.
+      if (!isset($document['id'])) {
+        $this->getLogger()->notice('Undefined FAO document ID, skipping document import.');
+        continue;
+      }
+      $id = $document['id'];
+      $import_record['imported_item_id'] = $id;
+
+      if (isset($manually_posted[$id])) {
+        $this->getLogger()->notice(strtr('FAO document @id already manually posted as report @report_id.', [
+          '@id' => $id,
+          '@report_id' => $manually_posted[$id],
+        ]));
+        continue;
+      }
+
+      // Retrieve the document URL.
+      if (!isset($document['documentLink'])) {
+        $this->getLogger()->notice(strtr('Undefined document URL for FAO document ID @id, skipping document import.', [
+          '@id' => $id,
+        ]));
+        continue;
+      }
+      $url = $document['documentLink'];
+      $import_record['imported_item_url'] = $url;
+
+      // Generate the UUID for the document.
+      $uuid = $this->generateUuid($url);
+      $import_record['imported_item_uuid'] = $uuid;
+
+      // Merge with existing record if available.
+      if (isset($existing_import_records[$uuid])) {
+        $import_record = $existing_import_records[$uuid] + $import_record;
+      }
+
+      $this->getLogger()->info(strtr('Processing FAO document @id.', [
+        '@id' => $id,
+      ]));
+
+      // Generate a hash from the data we use to import the document. This is
+      // used to detect changes that can affect the document on ReliefWeb.
+      $filtered_document = $this->filterArrayByKeys($document, [
+        'id',
+        'title',
+        'created',
+        'modified',
```

Comment:
I think modified should not be included in the data used to generate the hash because this may be updated for properties that should not trigger a re-import.